spec/fixtures/https_server.lua

```diff
@@ -0,0 +1,235 @@
+local https_server = {}
+https_server.__index = https_server
+
+
+local fmt = string.format
+local nginx_tpl_file = require 'spec.fixtures.nginx_conf_template'
```

Comment:
I wonder if another name like "spec.fixtures.balancer_nginx_conf_template" is better, because this name on the first look seems like a Kong template and may be confusing.

Reply:
> is there a reason to use another nginx instance and not the one in Kong itself using the server block injection?

I decided to use a different Nginx instance for some reasons:
- can be started and finished independently, easier to simulate failures.
- has a different error.log that can be easily parsed.
- we are sure it will never cause any side effects to the Kong instance.

None of these is a strong reason, and I can change that if you think it would be better to reuse the same instance. (@Tieske @dndx)

Reply:
> can be started and finished independently

is actually a good reason.

Regarding the API to use this thing, that should be exposed through `spec/helpers.lua` that's where all the generic test stuff resides (and is documented).

Reply:
Good catch! I will change that.